### PR TITLE
ch4/generic: fix anonymous union in MPIDIG request

### PR DESCRIPTION
**Pull request by yfguo**
## Pull Request Description

The MPIDIG request uses anonymous union which requires C11 support. This PR removes this usage and with a named union.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commits are self-contained and do not do two things at once
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] You or your company has a signed contributor's agreement on file with Argonne
* [ ] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

**Comment by yfguo** [comment]
This is fairly large change that affects many files which could cause merge conflicts. Since no one has reported any problem with the anonymous union, I am not sure if we really want to do this change.

@raffenet feel free to defer this PR. There is no work that depends on this.

File: src/mpid/ch4/include/mpidpre.h [hunk 1]
```diff
@@ -183,7 +183,7 @@ typedef struct MPIDIG_req_ext_t {
         MPIDIG_get_req_t greq;
         MPIDIG_cswap_req_t creq;
         MPIDIG_acc_req_t areq;
-    };
+    } ext;
```
Comment (hzhou) [inline src/mpid/ch4/include/mpidpre.h:186]: This makes me wonder what does `ext` mean. I would much prefer simply name it `u`. We have 15 instances of using `u` for "anonymous-like" union usage, and we are used to it. I don't think we need introduce extra names unless we have to.

Comment (raffenet) [reply src/mpid/ch4/include/mpidpre.h:186]: Agreed. Let's use `u`.

Comment (yfguo) [reply src/mpid/ch4/include/mpidpre.h:186]: sure. What I worried about more is whether this is really necessary given that the impact is large.

**Comment by yfguo** [comment]
test:mpich/ch4/most

**Pull request closed without merging**